# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/firefox/sparc.patch
# Copyright (C) 2020 - 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- firefox-81.0.2/third_party/rust/authenticator/src/linux/ioctl_sparc64.rs.vanilla	(nonexistent)
+++ firefox-81.0.2/third_party/rust/authenticator/src/linux/ioctl_sparc64.rs	2020-10-16 11:37:13.522007639 +0200
@@ -0,0 +1,5 @@
+/* automatically generated by rust-bindgen */
+
+pub type __u32 = ::std::os::raw::c_uint;
+pub const _HIDIOCGRDESCSIZE: __u32 = 1074022401;
+pub const _HIDIOCGRDESC: __u32 = 1342457858;
--- firefox-81.0.2/third_party/rust/authenticator/src/linux/hidwrapper.rs.vanilla	2020-10-16 11:16:06.521981704 +0200
+++ firefox-81.0.2/third_party/rust/authenticator/src/linux/hidwrapper.rs	2020-10-16 11:40:10.098011254 +0200
@@ -46,3 +46,10 @@
 
 #[cfg(all(target_arch = "s390x", target_endian = "big"))]
 include!("ioctl_s390xbe.rs");
+
+#[cfg(target_arch = "sparc")]
+include!("ioctl_sparc.rs");
+
+#[cfg(target_arch = "sparc64")]
+include!("ioctl_sparc64.rs");
+
--- firefox-81.0.2/third_party/rust/authenticator/src/linux/ioctl_sparc.rs.vanilla	(nonexistent)
+++ firefox-81.0.2/third_party/rust/authenticator/src/linux/ioctl_sparc.rs	2020-10-16 11:38:52.534009666 +0200
@@ -0,0 +1,5 @@
+/* automatically generated by rust-bindgen */
+
+pub type __u32 = ::std::os::raw::c_uint;
+pub const _HIDIOCGRDESCSIZE: __u32 = 1074022401;
+pub const _HIDIOCGRDESC: __u32 = 1342457858;
--- firefox-81.0.2/third_party/rust/authenticator/build.rs.vanilla	2020-10-15 23:53:32.177143375 +0200
+++ firefox-81.0.2/third_party/rust/authenticator/build.rs	2020-10-16 11:39:41.530010669 +0200
@@ -45,6 +45,10 @@
         "ioctl_aarch64be.rs"
     } else if cfg!(all(target_arch = "s390x", target_endian = "big")) {
         "ioctl_s390xbe.rs"
+    } else if cfg!(target_arch = "sparc") {
+        "ioctl_sparc.rs"
+    } else if cfg!(target_arch = "sparc64") {
+        "ioctl_sparc64.rs"
     } else {
         panic!("architecture not supported");
     };
